persistence-api/src/main/java/io/stargate/db/query/builder/Value.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright DataStax, Inc. and/or The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db.query.builder;
+
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
+import io.stargate.db.query.TypedValue;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/** A concrete value, or a bind marker. */
+public abstract class Value<T> {
+  private int internalIndex = -1;
+
+  public static <T> Value<T> marker() {
+    return new Marker<>();
+  }
```

Comment:
I've asked myself that same question. I think ultimately, we may have to if we want to get rid of query strings for the `Persistence` API. But most of the uses of `QueryBuilder` don't even bother with manually providing markers themselves, so I figured this could easily be a concern for later.